src/coreclr/jit/llvm.cpp

```diff
@@ -1685,6 +1684,43 @@ Value* Llvm::localVar(GenTreeLclVar* lclVar)
     return llvmRef;
 }
 
+void Llvm::buildLocalField(GenTreeLclFld* lclFld)
+{
+    unsigned int lclNum = lclFld->GetLclNum();
+
+    Value*     structAddrValue;
+    LclVarDsc* varDsc = _compiler->lvaGetDesc(lclNum);
+
+    if (canStoreLocalOnLlvmStack(varDsc))
+    {
+        assert(m_allocas[lclNum] != nullptr);
+
+        structAddrValue = m_allocas[lclNum];
+    }
+    else
+    {
+        structAddrValue = getLocalVarAddress(lclFld);
+    }
+
+    // TODO-LLVM: if this is an only value type field, or at offset 0, we can optimize
+    Value* structAddrInt8Ptr = castIfNecessary(structAddrValue, Type::getInt8PtrTy(_llvmContext));
+    Value* fieldAddressValue = _builder.CreateGEP(structAddrInt8Ptr, _builder.getInt16(lclFld->GetLclOffs()));
+
+    Type* fieldType;
+    if (lclFld->TypeGet() == TYP_STRUCT)
+    {
+        fieldType = getLlvmTypeForStruct(_compiler->lvaGetDesc(lclFld)->GetLayout());
```

Comment:
I think you mentioned that this was not goting to work for field, but I wasn't sure what to replace it with so have left it here for review.

Reply:
Yeah, the problem is that `LCL_FLD` does not support `TYP_STRUCT` (no way to get the layout out of the node). It's a very large IR issue I am working on right now.

So right now `buildLocalField` should just `assert(!lclFld->TypeIs(TYP_STRUCT);`.